script/lib/patches.py

```diff
@@ -0,0 +1,80 @@
+# import yaml  # TODO: Use for PatchesConfig.
+
+from git import apply as git_apply
+
+
+class Patch:
+  def __init__(self, file_path, repo_path):
+    self.file_path = file_path
+    self.repo_path = repo_path
+
+  def apply(self, reverse=False):
+    return git_apply(self.repo_path, self.file_path, reverse=reverse)
+
+  def reverse(self):
+    return self.apply(reverse=True)
+
+  def get_file_path(self):
+    return self.file_path
+
+
+class PatchesList:
+  def __init__(self, patches):
+    self.patches = patches
+
+  def apply(self, reverse=False, stop_on_error=True):
+    all_patches_applied = True
+    failed_patches = []
+
+    for patch in self.patches:
+      applied_successfully = patch.apply(reverse=reverse)
+
+      if not applied_successfully:
+        all_patches_applied = False
+        failed_patches.append(patch)
+
+      should_stop_now = not applied_successfully and stop_on_error
+      if should_stop_now:
+        break
+
+    return (all_patches_applied, failed_patches)
+
+  def reverse(self, stop_on_error=True):
+    return self.apply(reverse=True, stop_on_error=stop_on_error)
+
+
+class PatchesConfig:
+  def __init__(self, path):
+    raise
```

Comment:
I'm not familiar with this idiom; is this a way of marking the class as a WIP?

Reply:
Maybe ) I almost never use Python.
The whole `PatchesConfig` class wan't expected to be reviewed here, I should have make it clear. Or even exclude it from the PR altogether.